lib/ldapauth.js

```diff
@@ -288,7 +289,15 @@ LdapAuth.prototype._findGroups = function(user, callback) {
     return callback("no user");
   }
 
-  var searchFilter = self.opts.groupSearchFilter.replace(/{{dn}}/g, user[self.opts.groupDnProperty]);
+  var searchFilter = '';
+
+  if ((typeof (self.opts.groupSearchFilter) === 'string')) {
```

Comment:
Could you instead of this setup a function in the constructor, eg.

```javascript
if (typeof opts.groupSearchFilter === 'function') {
  this.groupSearchFilter = opts.groupSearchFilter;
} else {
  this.groupSearchFilter = function(user) {
    return opts.groupSearchFilter.replace(...);
  };
}
```

Then the authentication function would just do `var searchFilter = self.groupSearchFilter(user);`. I think it would look cleaner that way.